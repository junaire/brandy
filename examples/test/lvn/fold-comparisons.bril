# ARGS: -f

@main(arg1: int, arg2: int) {
  a: int = const 4;
  b: int = const 3;
  constant_fold2: bool = eq a b;
  constant_fold3: bool = le a b;
  constant_fold4: bool = lt b a;
  constant_fold5: bool = gt b a;
  constant_fold6: bool = ge b a;

  should_fold1: bool = eq arg1 arg1;
  should_fold2: bool = le arg1 arg1;
  should_fold3: bool = ge arg1 arg1;

  no_fold1: bool = eq arg1 arg2;
  no_fold2: bool = le arg1 arg2;
  no_fold3: bool = ge arg1 arg2;

  no_fold4: bool = lt arg1 arg1;
  no_fold5: bool = gt arg2 arg2;
}
