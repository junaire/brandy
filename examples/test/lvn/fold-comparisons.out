@main(arg1: int, arg2: int) {
  a: int = const 4;
  b: int = const 3;
  constant_fold2: bool = const false;
  constant_fold3: bool = const false;
  constant_fold4: bool = const true;
  constant_fold5: bool = const false;
  constant_fold6: bool = const false;
  should_fold1: bool = const true;
  should_fold2: bool = const true;
  should_fold3: bool = const true;
  no_fold1: bool = eq arg1 arg2;
  no_fold2: bool = le arg1 arg2;
  no_fold3: bool = ge arg1 arg2;
  no_fold4: bool = lt arg1 arg1;
  no_fold5: bool = gt arg2 arg2;
}
